### PR TITLE
Remove UnspecializedNumpyVariable - stop special casing numpy

### PR DESCRIPTION
**Pull request by voznesenskym**

### CODE REVIEW

**Comment by anijain2305** [comment]
What is the context behind it? Is specializing on numpy integers causing too many problems?

File: torchdynamo/variables/tensor.py [hunk 1]
```diff
@@ -646,42 +646,16 @@ def inline_torch_function_unwrapped(
...
-
 class UnspecializedPythonVariable(TensorVariable):
     """
     This is a 1-element tensor represents unspecialized python float/int.
     """
 
     def __init__(self, proxy: torch.fx.Proxy, **kwargs):
         raw_value = kwargs.pop("raw_value", None)
+        if isinstance(raw_value, np.number):
+            raw_value = raw_value.item()
```
Comment (anijain2305) [inline torchdynamo/variables/tensor.py:657]: Ah, got the point. Nice!

**Comment by voznesenskym** [comment]
> What is the context behind it? Is specializing on numpy integers causing too many problems?

No, just cleaning up code we don't need.

**Review by anijain2305 — APPROVED** [review]
+100

**Pull request merged**